Sources/NIOCore/EventLoopFuture.swift

```diff
@@ -386,31 +386,33 @@ public final class EventLoopFuture<Value> {
     internal var _callbacks: CallbackList
 
     @inlinable
-    internal init(_eventLoop eventLoop: EventLoop, value: Result<Value, Error>?, file: StaticString, line: UInt) {
+    internal init(_eventLoop eventLoop: EventLoop, value: Result<Value, Error>?) {
         self.eventLoop = eventLoop
         self._value = value
         self._callbacks = .init()
+    }
+
+    @inlinable
+    internal init(_eventLoop eventLoop: EventLoop, file: StaticString, line: UInt) {
+        self.eventLoop = eventLoop
+        self._value = nil
+        self._callbacks = .init()
 
         debugOnly {
             eventLoop._promiseCreated(futureIdentifier: _NIOEventLoopFutureIdentifier(self), file: file, line: line)
         }
     }
 
-    @inlinable
-    internal convenience init(_eventLoop eventLoop: EventLoop, file: StaticString, line: UInt) {
-        self.init(_eventLoop: eventLoop, value: nil, file: file, line: line)
-    }
-
     /// A EventLoopFuture<Value> that has already succeeded
     @inlinable
-    internal convenience init(eventLoop: EventLoop, value: Value, file: StaticString, line: UInt) {
-        self.init(_eventLoop: eventLoop, value: .success(value), file: file, line: line)
+    internal convenience init(eventLoop: EventLoop, value: Value) {
+        self.init(_eventLoop: eventLoop, value: .success(value))
     }
 
     /// A EventLoopFuture<Value> that has already failed
     @inlinable
-    internal convenience init(eventLoop: EventLoop, error: Error, file: StaticString, line: UInt) {
-        self.init(_eventLoop: eventLoop, value: .failure(error), file: file, line: line)
+    internal convenience init(eventLoop: EventLoop, error: Error) {
+        self.init(_eventLoop: eventLoop, value: .failure(error))
```

Comment:
Can I propose that instead of using a `convenience init` here we make these proper `init`s? That will reduce the temptation to use the init that doesn't do file and line storage, as you either have a value up front or you don't.